File: Python3/SelfDividingNumbers.py

left = 1
right = 22

#Output: [1, 2, 3, 4, 5, 6, 7, 8, 9, 11, 12, 15, 22]

ans = True
final = []
for i in range(left,right+1):   #可以把left直接換成i 就好了
    for y in range(0,len(str(left))):

        if int(str(left)[y]) == 0:
            ans = False
        elif left % int(str(left)[y]) != 0 :
            ans = False

    if ans == True:
        final.append(left)
    left = left + 1
    ans = True


print(final)

File: Java/SelfDividingNumbers.java
import java.util.ArrayList;

public class SelfDividingNumbers {
     public static void main(String[] args) {

         int left = 1;
         int right = 22;

        //Output: [1, 2, 3, 4, 5, 6, 7, 8, 9, 11, 12, 15, 22]

         boolean ans  = true;
         ArrayList A = new ArrayList();
         int mod;
         int cal;
         for(int i = left;i<=right;i++){
             cal = i;
             for(int j = 0;j<String.valueOf(i).length();j++){
                 mod = cal % 10;
                 cal = cal / 10;
                 if(mod == 0){
                    ans = false;
                 }else if(i % mod != 0){
                     ans = false;
                 }
             }

             if(ans==true){
                 A.add(i);
             }
             ans = true;
         }


         System.out.print(A);




    }
}
